python/pyspark/sql/catalog.py

```diff
@@ -255,9 +255,26 @@ def registerFunction(self, name, f, returnType=StringType()):
         >>> _ = spark.udf.register("stringLengthInt", len, IntegerType())
         >>> spark.sql("SELECT stringLengthInt('test')").collect()
         [Row(stringLengthInt(test)=4)]
```

Comment:
Let's fix the doc for this too. It says `:param f: python function` but we could describe that it takes Python native function, wrapped function and `UserDefinedFunction` too.

Reply:
ok